final_html_output.py

```diff
@@ -11,14 +11,26 @@ def final_html_output(df: pd.DataFrame) -> str:
         html string
     """
     html_string = """
```

Comment:
Can you move the html from here to a separate html file - eg: `assets/index.html`. This way you can format and update it   properly.

And you can read the file here as string the same way.